### PR TITLE
In add test duplicate title error is disappearing

### PR DESCRIPTION
**Pull request by yash6102000**
fix:In add test duplicate title error is disappearing
along with name and description error
after adding a new test there is not delay to redirect

### CODE REVIEW

File: app/components/sections/AddEditSection.tsx [hunk 1]
```diff
@@ -54,8 +58,60 @@ const AddEditSection = ({
...
+      navigate(`${location.pathname}${location.search}`)
+      setOpen(false)
+    }
+  }, [
+    fetcherData?.resp?.data?.id,
+    fetcherData?.resp?.status,
+    data?.filters,
+    navigate,
+    setOpen,
+
```
Comment (Dhruv-Copods) [inline app/components/sections/AddEditSection.tsx:99]: remove extra empty space

File: app/routes/$workspaceId/tests.tsx [hunk 2]
```diff
@@ -74,16 +74,50 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
+      if (Object.values(createSectionFieldError).some(Boolean)) {
+        return json({ createSectionFieldError }, { status: 400 })
+      }
+      const response = await handleAddTest(
+        name as string,
+        description as string,
+        createdById,
+        currentWorkspaceId
+      )
+      return response
```
Comment (Dhruv-Copods) [inline app/routes/$workspaceId/tests.tsx:118]: we can directly return

File: app/interface/Interface.tsx [hunk 3]
```diff
@@ -284,8 +284,9 @@ export interface CandidateResult {
   link?: string
 }
 export interface sectionActionErrorsType {
```
Comment (anuragP04) [inline app/interface/Interface.tsx:286]: can we remove this from interface file.

File: app/components/sections/AddEditSection.tsx [hunk 4]
```diff
@@ -54,8 +58,59 @@ const AddEditSection = ({
     )
   }
   const addSection = (name: string, description: string) => {
-    submit({ addSection: "sectionAdd", name, description }, { method: "post" })
+    fetcher.submit(
+      { addSection: "sectionAdd", name, description },
+      { method: "get" }
```
Comment (anuragP04) [inline app/components/sections/AddEditSection.tsx:63]: why you have used method get here?

File: app/routes/$workspaceId/tests.tsx [hunk 5]
```diff
@@ -74,16 +74,49 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
+      }
+    }
+
+    const validateDescription = (description: string) => {
+      if (typeof description !== "string" || description.length <= 0) {
+        return "statusCheck.descIsReq"
+      }
+    }
+
+    if (action === "sectionAdd") {
```
Comment (anuragP04) [inline app/routes/$workspaceId/tests.tsx:101]: why this functionality is in loader not in action?

File: app/components/sections/AddEditSection.tsx [hunk 6]
```diff
@@ -54,8 +58,59 @@ const AddEditSection = ({
...
+  }, [
+    fetcherData?.resp?.data?.id,
+    fetcherData?.resp?.status,
+    data?.filters,
+    navigate,
+    setOpen,
+    t,
+    fetcherData?.resp?.status,
+  ])
+  useEffect(() => {
```
Comment (anuragP04) [inline app/components/sections/AddEditSection.tsx:102]: what is the use of this useEffect here?

File: app/components/sections/AddEditSection.tsx [hunk 7]
```diff
@@ -54,8 +58,59 @@ const AddEditSection = ({
...
+    if (fetcherData?.createSectionFieldError || fetcherData?.errors?.title) {
+      setSectionActionErrors({
+        title: fetcherData?.createSectionFieldError?.title,
+        description: fetcherData?.createSectionFieldError?.description,
+        duplicateTitle: fetcherData?.errors?.title,
+      })
+    }
+  }, [fetcherData?.createSectionFieldError, setSectionActionErrors])
+
+  const navigate = useNavigate()
```
Comment (anuragP04) [inline app/components/sections/AddEditSection.tsx:78]: what is this useEffect doing here?

File: app/components/sections/AddEditSection.tsx (hunk 6 above)
Comment (yash6102000) [reply app/components/sections/AddEditSection.tsx:102]: if there is an error like name is required then if start typing then input field is not empty so error is gone

File: app/components/sections/AddEditSection.tsx (hunk 7 above)
Comment (yash6102000) [reply app/components/sections/AddEditSection.tsx:78]: in case of succes show toster and close the popup

File: app/routes/$workspaceId/tests.tsx (hunk 5 above)
Comment (yash6102000) [reply app/routes/$workspaceId/tests.tsx:101]: adding this in post

File: app/interface/Interface.tsx (hunk 3 above)
Comment (yash6102000) [reply app/interface/Interface.tsx:286]: yes..then we have to defined the type at where props are recived

Comment (anuragP04) [reply app/interface/Interface.tsx:286]: then remove it

**Review by anuragP04 — APPROVED** [review]
